### ISSUE TITLE
Problem with fixed+persistant footers & headers?

### ISSUE THREAD

**Issue by ALuserX** [ToddThomson/jQuery-Mobile-Subpage-Widget#27]
Hello Todd,

first of all thanks for a great widget.

I am now working (or trying to work) on a web-app and use JQM for the first time. It should have many different pages, with many different subpages each, and since it's not possible to do in JQM without disabling ajax navigation I am really happy that there is such a widget.

But, there is one problem though. I use persistent fixed headers and footers on each page, that should give user a feeling that he stays on the same page all the time. As per default in JQM user always can click on the content to dismiss both fixed header & footer.

And somehow it seems that when I try to open a subpage header & footer stop working as they should. They overlap the content and stay fixed all the time, even if I try to dismiss them by clicking on the content.

So far I only tried it with desktop versions of FF and GC, but with same results.

Here is a little demo I prepared:
http://aluserex.vshost.ru/jqm-subpage/

I hope you can reproduce my problem and know what the reason is or could be. Maybe I am just doing something wrong.

**Comment by ToddThomson** [comment]
@ALuserX Yes, I see that the page content is not being positioned properly below the header. This is prob. due to CSS class(es) not being assigned properly. Can you follow this up - look at the non-subpage page css with the browser inspector and compare it with the subpage css.

**Comment by kgx** [comment]
I am having the same issue in my application. This plugin is causing something goofy with JQM's ability to programmatically add top padding to the each page div involved when using fixed headers. It seems like it must be a sequencing issue... perhaps you can manipulate the DOM prior to whatever JQM process adds the padding? 

Here is my CSS workaround:

```
 #parent-page, #subpage, #other-subpage {
   padding-top: 70px !important;
 }
```

BTW Todd thanks for creating this plugin it is very helpful. I'm surprised that loading multiple pages via AJAX is not core functionality in JQM.

**Comment by OnlyByGrace** [comment]
This problem is created because the Fixed Toolbar widget is responsible for setting the proper padding attributes on it's parent page. But it determines it's parent page by looking for the closest element with the "ui-page" class. But when the subpage's fixed toolbar is processed the subpage has not been processed, so it doesn't have the ui-page class. The toolbar sets it's parent as the parent page.

In short, you can fix this by adding a class to your subpage:

```
<div data-role="subpage" class="ui-page">
</div>
```

**Comment by OnlyByGrace** [comment]
On a second look, the problem is that all default jQuery Mobile widgets are being processed before subpage gets the opportunity to process itself. This was creating problems for me with lots of different widgets (for example, my panel was binding itself to the parent page events instead of the subpage events, making it impossible for me to open it when a href="#panelid" anchor was clicked).

So the problem was fixed more broadly by changing the subpage widget to use the pagebeforecreate event instead of the pagecreate event:

```
$(document).bind("pagebeforecreate create", function (e) {
    $.mobile.subpage.prototype.enhanceWithin(e.target);
});
```

http://jquerymobile.com/demos/1.2.0/docs/api/events.html

**Comment by ToddThomson** [comment]
Thanks @OnlyByGrace , I'm starting an upgrade project which involves the subpage widget next week and I'll test your fix then.

**Comment by ToddThomson** [comment]
Thanks @OnlyByGrace . I've tested and committed your fix.
